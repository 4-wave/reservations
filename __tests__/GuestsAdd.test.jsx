import React from 'react';
import renderer from 'react-test-renderer';
import GuestAdd from '../src/Components/GuestsAdd.jsx';
import { shallow } from 'enzyme';
import { create } from "react-test-renderer";

describe('<GuestAdd /> component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<GuestAdd />);

        expect(tree).toMatchSnapshot();
    });

    it('initiates with state of 0 children', () => {
        const wrapper = shallow(<GuestAdd />);
        expect(wrapper.state().childrenCount).toBe(0);
    });
    it('initiates with state of 0 children', () => {
        const wrapper = shallow(<GuestAdd />);
        expect(wrapper.state().adultsCount).toBe(1);
    });
    it('initiates with state of 0 children', () => {
        const wrapper = shallow(<GuestAdd />);
        expect(wrapper.state().infantsCount).toBe(0);
    });

    it('initiates with test', () => {
        const wrapper = shallow(<GuestAdd />);
        expect(wrapper.contains("4 guests maximum. Infants donâ€™t count toward the number of guests.")).toBe(true)
    });
    
    it('changes the state for adultCount', () => {
        const props = { countAdults: () => {}}
        const wrapper = create(<GuestAdd {...props}/>);
        const instance = wrapper.getInstance();
        expect(instance.state.adultsCount).toBe(1);
        instance.adultsCountUp();
        expect(instance.state.adultsCount).toBe(2);
        instance.adultsCountDown();
        expect(instance.state.adultsCount).toBe(1);
    });

    it('changes the state for children count', () => {
        const props = { countChildren: () => {}}
        const wrapper = create(<GuestAdd {...props}/>);
        const instance = wrapper.getInstance();
        expect(instance.state.childrenCount).toBe(0);
        instance.childrenCountUp();
        expect(instance.state.childrenCount).toBe(1);
        instance.childrenCountDown();
        expect(instance.state.childrenCount).toBe(0);
    });

    it('changes the state for infants count', () => {
        const props = { countInfants: () => {}}
        const wrapper = create(<GuestAdd {...props}/>);
        const instance = wrapper.getInstance();
        expect(instance.state.infantsCount).toBe(0);
        instance.infantsCountUp();
        expect(instance.state.infantsCount).toBe(1);
        instance.infantsCountDown();
        expect(instance.state.childrenCount).toBe(0);
    });
});