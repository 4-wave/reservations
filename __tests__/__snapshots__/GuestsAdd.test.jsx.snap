// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuestAdd /> component matches the snapshot 1`] = `
<div
  className="sc-bdVaJa iBvhcS"
>
  <div
    className="sc-bwzfXH kRaKT"
  >
    <div
      className="sc-htpNat gZwPAN"
    >
      <div
        className="sc-bxivhb FkGqL"
      >
        Adults
      </div>
    </div>
    <div>
      <button
        className="sc-EHOje bTSweX"
        disabled={true}
        onClick={[Function]}
      >
         - 
      </button>
        
      1
        
      <button
        className="sc-EHOje bTSweX"
        disabled={false}
        onClick={[Function]}
      >
         + 
      </button>
    </div>
  </div>
  <div
    className="sc-bwzfXH kRaKT"
  >
    <div
      className="sc-htpNat gZwPAN"
    >
      <div
        className="sc-bxivhb FkGqL"
      >
        Children
      </div>
      <div
        className="sc-ifAKCX jTvgFe"
      >
        Ages 2-12
      </div>
    </div>
    <div>
      <button
        className="sc-EHOje bTSweX"
        disabled={true}
        onClick={[Function]}
      >
         - 
      </button>
        
      0
        
      <button
        className="sc-EHOje bTSweX"
        disabled={false}
        onClick={[Function]}
      >
         + 
      </button>
    </div>
  </div>
  <div
    className="sc-bwzfXH kRaKT"
  >
    <div
      className="sc-htpNat gZwPAN"
    >
      <div
        className="sc-bxivhb FkGqL"
      >
        Infants
      </div>
      <div
        className="sc-ifAKCX jTvgFe"
      >
        Under 2
      </div>
    </div>
    <div>
      <button
        className="sc-EHOje bTSweX"
        disabled={true}
        onClick={[Function]}
      >
         - 
      </button>
        
      0
        
      <button
        className="sc-EHOje bTSweX"
        disabled={false}
        onClick={[Function]}
      >
         + 
      </button>
    </div>
  </div>
  <div
    className="sc-bwzfXH kRaKT"
  >
    <div
      className="sc-ifAKCX jTvgFe"
    >
      4 guests maximum. Infants donâ€™t count toward the number of guests.
    </div>
  </div>
</div>
`;
